src/HttpLib/protocol/Protocol.java

```diff
@@ -0,0 +1,16 @@
+package HttpLib.protocol;
+
+import HttpLib.Exceptions.InvalidRequestException;
+import HttpLib.Exceptions.InvalidResponseException;
+import HttpLib.HttpRequest;
+import HttpLib.IRequestCallback;
+
+import java.io.IOException;
+
+public interface Protocol {
```

Comment:
Love it! Would name it IProtocol thought

Reply:
done!